### DISCUSSION TITLE
MiBand (5+6) integration

### DISCUSSION THREAD

**Discussion by elgrupo123** [NightscoutFoundation/xDrip#3097] [Ideas]
The direct integration of the Xiaomi MiBand series up to the 4th generation is really very successful. Respect! 
Can we also expect MiBand 5 and 6 in the near future (without WatchDrip)? 
Greetings

**Comment by Navid200** [comment]
Why?

The whole point of Watchdrip was to outsource.
We don't have enough manpower to cover what xDrip has already.  We cannot add more.
There are features we have to add.  We really don't have enough developers.

Why would we include Miband now?

I want us to outsource WearOS now.  There is no reason to include Miband in xDrip now.

**Comment by elgrupo123** [comment]
It was just a question ;) 
Why? because you want to have as few apps and sources of error on your smartphone as possible. The problem with the lack of developers is understandable and I completely understand that. 
But it would be a shame if the app. slimmed down the range of functions.

Thank you for your great work!!!

**Reply by Navid200** [reply]
That makes sense

You may not be aware of the history.
As more and more used MiBand, more and more bugs and feature requests were added here.
But, they were not moving forward.  You can see that we have a backlog of PRs.  It takes time to review them.  Everything needs to be tested to verify that the PR will not break anything.  Looking at a PR, it may seem that all needed is to click on merge.  But, a lot goes on behind the scenes.  A lot of tests are performed by the owner of the repository that none of us ever is aware of.  But, they take a lot of time.
  
We suggested the addition of a companion app.
The con of the companion app, as you have pointed out, is that the user will need to install one more app.
The pro is that anything related to MiBand, bugs and features, can be added in the companion app.  Occasionally, there may be a need for a small tweak to the service that provides the communications link between xDrip and the companion app.
The result is that new features can be added much easier.

Since the companion app has been created, users have been happily using MiBand devices.
I consider that a successful compromise for the users.

It's not easy to know what the background of the person asking a question is in a forum.  It's also pointless to go over all of the details as soon as a question is asked.
So, it is needed to have a conversation (like "why") in order to find out what needs to be discussed.

I hope that makes sense.

**Reply by mcassiopeia** [reply]
Hello,

the reason why the direct connection without Watchdrip is requested is in my view very simple: Whatchdrip is very unstable. 

It crashes very often and you have to restart the connection to the band. Sometimes you even have to restart the phone in order to get it connected again. 
When it sends data it is delayed for at least a minute and while you recharge the phone it does not sent any data at all.

When I start the battery maintanance of my phone it always shows me a sad smiley that tells me that there is a problem with the battery and suggestes to put Watchdrip in the deep sleep mode because it crashes too often. But that of cause is not possible. 

Please do not get me wrong. I am very grateful for all the work you put in these programs and I understand if you say, that it is not possible to include the bands in the direct connection with xdrip but please do not ask why somebody would wish for it...

**Reply by elgrupo123** [reply]
I agree. Ultimately, Xiaomi and Zepp Life are total crap and extreme China surveillance. But for the kids, blood sugar monitoring is a good thing and you're happy when it runs smoothly without the problems mentioned, which I can confirm.

**Reply by Navid200** [reply]
Have you guys discussed your criticism of Watchdrip with its developer?
